packages/main/src/components/protected/sidebar/components/DropRoom.jsx

```diff
@@ -1,10 +1,26 @@
-import React from "react";
+import React, { useState } from "react";
 import styles from "../styles/Drop.module.css";
 import ToggleSwitch from "./ToggleSwitch";
 import { BsToggleOff } from "react-icons/bs";
 import GrCircleAlert from "react-icons/gr";
+import useChannel from "../../../../auth/useChannel";
 
 const DropRoom = props => {
+  const { updatePrivacy, createChannel } = useChannel(props.data?.roomId);
+  const [state, setState] = useState({});
+
+  const handleEvent = ({ target }) => {
+    setState({
+      ...state,
+      [target.name]: target.type === "checkbox" ? target.checked : target.value
```

Comment:
setState(prev => ({ ..prev, then update the values of what you need to change.}))